Shared/sdk/SharedUtil.Crypto.h

```diff
@@ -25,18 +26,66 @@ namespace SharedUtil
         SString publicKey, privateKey;
     };
 
-    inline SString Base64encode(const SString& data)
+    inline SString Base64encode(const SString& data, const SString& variant = SString())
     {
-        SString                result;
-        CryptoPP::StringSource ss(data, true, new CryptoPP::Base64Encoder(new CryptoPP::StringSink(result), false));            // Memory is freed automatically
+        SString result;
+
+        if (variant == "URL")
```

Comment:
Okay, we came to the conclusion that a string representation is better then enum in this particular case. I would just suggest a tiny improvement: move `"HEX"` and `"URL"` literals into an unnamed namespace and make them `const std::string`. In this way, we will be able to eliminate calculation of string length every time.

Reply:
To avoid breaking NRVO, they should remain SStrings, but I've moved them to a separate namespace as constants.

Reply:
In the comparison statements the NRVO doesn't exist because it has no returning values(at least within the bounds of an expression). So there is no need to keep using SString. But SString is also okay, fine.

Reply:
Okay, didn't know about that it only applies if the return is inside an expression.

Reply:
Other arguments are SStrings too, i didn't wanna mix them tbh

Reply:
Oh, I didn't notice that those functions are within a header file. In this case unnamed namespace is unacceptable. So you can either reverse the last commit or create SharedUtil.Crypto.hpp and put the functions(along with the namespace) into it (you can see how the hpp's are included in SharedUtil.cpp). Once this is done I approve the PR.

Reply:
> Oh, I didn't notice that those functions are within a header file. In this case unnamed namespace is unacceptable. So you can either reverse the last commit or create SharedUtil.Crypto.hpp and put the functions(along with the namespace) into it (you can see how the hpp's are included in SharedUtil.cpp). Once this is done I approve the PR.

Done.
@tederis